sourcegraph/sourcegraph
# Add prefix matching for global defs search

Use `ts_query prefix` technic to do prefix matching on global defs. SQL tested on production database all finished within 50ms.

Before:

![image](https://cloud.githubusercontent.com/assets/2946214/15790784/19c2021a-2988-11e6-9782-adc47d35212e.png)

After:

![image](https://cloud.githubusercontent.com/assets/2946214/15791215/d0e4298a-298a-11e6-96e3-eee5542a6618.png)

##### Test plan

1. Search any token that does not have exact match, should always return something (if have prefix matched results)
2. Search token does have exact match, prefix matched results will have relatively lower ranking, so exact matches are on the top.
